src/cargo/util/network.rs

```diff
@@ -0,0 +1,46 @@
+use util::{CargoResult, human, Config, errors};
+use term::color::YELLOW;
+
+/// Wrapper method for network call retry logic.
+///
+/// Retry counts provided by Config object 'net.retry'. Config shell outputs
+/// a warning on per retry.
+///
+/// Closure must return a CargoResult.
+///
+/// Example:
+/// use util::network;
+/// cargo_result = network.with_retry(&config, || something.download());
+pub fn with_retry<T, E, F>(config: &Config, mut callback: F) -> CargoResult<T>
+    where F: FnMut() -> Result<T, E>,
+          E: errors::NetworkError + errors::CargoError
+{
+    let net_retry = try!(config.net_retry());
+    let mut result: CargoResult<T> = Err(human("failed to initialize network call"));
+    for try in 0..net_retry + 1 {
+        result = callback();
+        if result.is_err() {
```

Comment:
What do you see maybe_spurious doing? I am thinking a is_retryable makes more sense. Then it would be `if result.is_err() && result.is_retryable()` which reads like I am thinking the code works. Right now it would return true for the two errors but it could do more for other errors.